src/swap/hooks/useAwaitCalls.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback } from 'react';
+import { waitForTransactionReceipt } from 'wagmi/actions';
+import { useCallsStatus } from 'wagmi/experimental';
+import type { UseAwaitCallsParams } from '../types';
+
+export function useAwaitCalls({
+  accountConfig,
+  config,
+  lifecycleStatus,
+  setLifecycleStatus,
+}: UseAwaitCallsParams) {
+  const callsId =
+    lifecycleStatus.statusName === 'transactionApproved'
+      ? lifecycleStatus.statusData?.callsId
+      : undefined;
+
+  const { data } = useCallsStatus({
+    id: callsId || '',
+    query: {
+      refetchInterval: (query) => {
+        return query.state.data?.status === 'CONFIRMED' ? false : 1000;
+      },
+      enabled: callsId !== undefined,
+    },
+  });
+
+  return useCallback(async () => {
+    if (data?.status === 'CONFIRMED' && data?.receipts) {
+      const transactionReceipt = await waitForTransactionReceipt(
+        accountConfig,
+        {
+          confirmations: 1,
+          hash: data.receipts[data.receipts.length - 1].transactionHash,
+        },
+      );
+      setLifecycleStatus({
+        statusName: 'success',
+        statusData: {
+          isMissingRequiredField: false,
+          maxSlippage: config.maxSlippage,
```

Comment:
i don't think you should need to pass these shared fields

Reply:
these are required in the typing - should we make them optional?

cc: @cpcramer